polygeist/tools/cgeist/driver.cc

```diff
@@ -1164,6 +1165,58 @@ template <typename T> static void filterFunctions(T Module) {
     Func.erase();
 }
 
+static int readHostModule(MLIRContext &context, ModuleOp module,
+                          llvm::StringRef inputFilename) {
+  Block &dest = module.getBodyRegion().front();
+  assert(dest.getOperations().size() == 1 &&
+         "Expecting a single operation in the original module");
+
+  // Set up the input file.
+  std::string errorMessage;
+  auto file = openInputFile(inputFilename, &errorMessage);
+  if (!file) {
+    llvm::errs() << errorMessage << "\n";
+    return -1;
+  }
+
+  // Tell sourceMgr about this buffer, which is what the parser will pick up.
```

Comment:
Mostly copied from [this file](https://github.com/intel/llvm/blob/b11316ad6a8b8c2d6537706ccdcc3a77f8e17c4e/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp)